### PR TITLE
Feat/react router dom update

### PR DESCRIPTION
**Pull request by NullDivision**
- add support for flow v0.53.x for react-router-dom v4 package

### CODE REVIEW

File: definitions/npm/react-router-dom_v4.x.x/flow_v0.53.x/react-router-dom_v4.x.x.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    exact?: bool,
+    strict?: bool
+  }> {}
+
+  declare export class Switch extends React$Component<{
+    children?: Array<React$Element<*>>
+  }> {}
+
+  declare type FunctionComponent<P> = (props: P) => ?React$Element<any>;
+  declare type ClassComponent<D, P, S> = Class<React$Component<D, P, S>>;
```
Comment (tirli) [inline definitions/npm/react-router-dom_v4.x.x/flow_v0.53.x/react-router-dom_v4.x.x.js]: Hi. It seems like you forgot to remove DefaultProps from React$Component

Comment (NullDivision) [inline definitions/npm/react-router-dom_v4.x.x/flow_v0.53.x/react-router-dom_v4.x.x.js]: Good point. Thank you.

**Comment by tirli** [comment]
Hi, I tried to copy your definitions to my project and it fails. I make it woks with [such definitions](https://gist.github.com/tirli/b2ca8ba23308d046d68e6744ac44f1ba). And I also added them to react-dom package as pointed in NOTE section. It is also should be added to react-dom-native. Maybe you'll merge our definitions to create correct PR?

**Comment by NullDivision** [comment]
Thank you @tirli. If you feel up to trying the latest push and see if it works I'd be grateful. If something goes wrong a code snippet would also be nice to add to the tests

File: definitions/npm/react-router-dom_v4.x.x/flow_v0.53.x/react-router-dom_v4.x.x.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    strict?: boolean
+  }> {}
+
+  declare export class Switch extends React$Component<{
+    children?: React$Node
+  }> {}
+
+  declare export function withRouter<P>(
+    Component: React$ComponentType<P>
+  ): React$ComponentType<$Diff<P, ContextRouter>>;
```
[the remark this answers is not in the document]
Comment (NullDivision) [reply definitions/npm/react-router-dom_v4.x.x/flow_v0.53.x/react-router-dom_v4.x.x.js]: Updated both definition and tests. You were right. You get a cookie. Thank you sir. :)

**Pull request merged**